mobile_app/upload_build.py

```diff
@@ -0,0 +1,70 @@
+"""
+Uploads an app build. Presently it only uploads to hockey app"
+"""
+
+from collections import namedtuple
+import logging
+import os
+import requests
+import sys
+
+from mobile_app.exceptions import (
+    MissingEnvironmentVariable,
+    UploadFailure
+)
+
+logger = logging.getLogger(__name__)  # pylint: disable=invalid-name
+
+UPLOAD_URL = "https://rink.hockeyapp.net/api/2/apps/upload"
+
+
+class HockeyTokenAuth(object):
+    """
+    Authorization method for requests library supporting HockeyApp tokens
+    """
+    def __init__(self, token):
+        self.token = token
+
+    def __call__(self, request):
+        request.headers["X-HockeyAppToken"] = self.token
+        return request
+
+
+def run_upload_build(environ):
+    """
+    Uploads a build to Hockey App
+
+    Arguments
+        params (dict): Arguments to the upload request. Will be passed
+            as form encoding fields
+    """
+    try:
+        token = environ["HOCKEY_APP_TOKEN"]
+        commit_sha = environ["CODE_SHA"]
+        binary_path = environ["BINARY_PATH"]
+    except KeyError as error:
+        raise MissingEnvironmentVariable(error.args[0])
+
+    with open(binary_path, "rb") as binary_file:
+        binary_data = binary_file.read()
+    auth = HockeyTokenAuth(token)
+    params = {
+        "commit_sha": commit_sha,
+    }
+
+    files = {
+        "ipa": (os.path.split(binary_path)[1], binary_data)
+    }
+
+    response = requests.post(UPLOAD_URL, data=params, auth=auth, files=files)
+    if response.status_code != 201:
+        raise UploadFailure(response)
```

Comment:
The upload part is actually a robot that runs on travis. It's uploading artifacts and I sort of expect they're bigger as time goes on, so I'm hesitant to pick any relatively short time and a long timeout isn't super useful - I'm sure travis shuts down the VM at some point anyway. A connection timeout would be reasonable, but again, I expect that if it can't actually connect, it means the service is down and it will fail pretty quickly.